tickit/adapters/epicsadapter.py

```diff
@@ -11,7 +11,7 @@
 from tickit.core.adapter import ConfigurableAdapter
 
 
-@dataclass
+@dataclass(eq=False)
```

Comment:
Why are we forbidding equality checks?

Reply:
Dataclasses are not hashable otherwise (see: https://docs.python.org/3/library/dataclasses.html). This class intends to use them as the key for: `interrupt_records: Dict[InputRecord, Callable[[], Any]]` so I assume not making them hashable is an oversight?

Reply:
I think we'd be better off making them immutable with `frozen=True` if we wish for them to be hashable as it would retain the ability to check for equality and make the hash safe.

Reply:
Yeah I think that does make more sense.